src/NUnitFramework/tests/Constraints/DictionaryContainsKeyConstraintTests.cs

```diff
@@ -189,7 +190,7 @@ public void ShouldCallContainsKeysMethodOnPlainDictionary()
         {
             var dictionary = new TestNonGenericDictionary(99);
 
-            Assert.Catch<ArgumentException>(() => Assert.That(dictionary, Does.ContainKey(99)));
```

Comment:
Note to reviewer: This test was previously detecting that we were throwing an exception when using an `IDictionary`. I've updated it to pass. I didn't add any additional tests because we already have "Should fail" tests in place for non-dictionaries:
https://github.com/nunit/nunit/blob/a5adf028c101283beecad0b49104468efe75b90f/src/NUnitFramework/tests/Constraints/DictionaryContainsKeyConstraintTests.cs#L68-L76